### PR TITLE
Update dependencies and test targets

### PR DESCRIPTION
**Pull request by westy92**
* Update TypeScript and other dependencies.
* Update test targets. Current `LTS` and `stable` are `14.x` and `15.x` (https://nodejs.org/en/about/releases/).
* Cleanup code; update imports to improve type hints and remove deprecations.

### CODE REVIEW

**Review by Niek — APPROVED** [review]

**Comment by Niek** [comment]
@patrickhulce any chance to merge this?

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,11 @@
 language: node_js
 matrix:
   include:
-    - node_js: "8"
```
Comment (patrickhulce) [inline .travis.yml:4]: let's leave node 8 in here until we officially drop support for it in our next major?

I'm with ya on dropping it, just don't want to publish a new major at this exact moment :)

File: package.json [hunk 2]
```diff
@@ -11,22 +11,22 @@
...
+    "mocha": "^8.2.1",
     "sinon": "^9.0.1",
-    "ts-node": "8.3.0",
-    "typescript": "3.9.2"
+    "ts-node": "^9.0.0",
+    "typescript": "^4.1.2"
   },
   "dependencies": {
     "@types/node": "*",
-    "escape-string-regexp": "^1.0.5",
```
Comment (patrickhulce) [inline package.json:26]: can't use 3+ without node 10 only

File: package.json [hunk 3]
```diff
@@ -11,22 +11,22 @@
...
+    "escape-string-regexp": "^4.0.0",
     "is-wsl": "^2.2.0",
     "lighthouse-logger": "^1.0.0",
-    "mkdirp": "^0.5.3",
```
Comment (patrickhulce) [inline package.json:29]: same, though it technically works at the moment.

**Review by patrickhulce — CHANGES_REQUESTED** [review]
thanks @westy92 all of your stated goals are great 👍 

just don't think we want to introduce breaking changes parts of this right now. maybe file an issue for the collection of breaking changes we'll want to batch for the next major? :)

**Comment by westy92** [comment]
I reverted the changes that require Node.js 10+. I'm happy to open another PR once this lands that drops Node.js 8.x support that can sit around until the project is ready for it.

File: src/chrome-launcher.ts [hunk 4]
```diff
@@ -304,7 +304,7 @@ class Launcher {
   waitUntilReady() {
     const launcher = this;
 
-    return new Promise((resolve, reject) => {
+    return new Promise<{}>((resolve, reject) => {
```
Comment (patrickhulce) [inline src/chrome-launcher.ts:307]: these could probably be better expressed as `Promise<void>` but it's not a big deal :)

**Review by patrickhulce — APPROVED** [review]
thanks for working on this @westy92 !

File: src/chrome-launcher.ts (hunk 4 above)
Comment (westy92) [reply src/chrome-launcher.ts:307]: I was going to, but these existing spots had `{}` so I left it:
https://github.com/GoogleChrome/chrome-launcher/blob/master/test/chrome-launcher-test.ts#L26
https://github.com/GoogleChrome/chrome-launcher/blob/master/src/chrome-launcher.ts#L50

Comment (patrickhulce) [reply src/chrome-launcher.ts:307]: heh, yeah those are mistakes IMO, but nbd

**Comment by westy92** [comment]
You're welcome! I addressed all the comments and it looks like all the tests passed. 🎉 Is there anything else I need to do to get this merged in?

**Comment by patrickhulce** [comment]
nope I was just waiting on the tests and then forgot about it :) thanks @westy92 !

**Comment by westy92** [comment]
> I'm happy to open another PR once this lands that drops Node.js 8.x support that can sit around until the project is ready for it.

Just to close the loop on this, here's my promised PR: https://github.com/GoogleChrome/chrome-launcher/pull/222

**Pull request merged**